lib/ServiceNameResolver.h

```diff
@@ -29,29 +29,33 @@ namespace pulsar {
 class ServiceNameResolver {
```

Comment:
Add a `#include <memory>` to this file for `std::make_shared`. The compilation did not fail because the `.cc` files that include this header already include `<memory>`.